src/parquet/column/writer.cc

```diff
@@ -171,7 +171,7 @@ void ColumnWriter::AddDataPage() {
 
 void ColumnWriter::WriteDataPage(const CompressedDataPage& page) {
   int64_t bytes_written = pager_->WriteDataPage(page);
-  total_bytes_written_ += bytes_written;
+  total_bytes_written_ += static_cast<int>(bytes_written);
```

Comment:
We should make `total_bytes_written_` either int32_t or int64_t. If it's 64-bit in the Thrift metadata, then int64_t